infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/pvc/CommonPVCStrategy.java

```diff
@@ -86,7 +91,11 @@ public void prepare(InternalEnvironment env, OpenShiftEnvironment osEnv, String
   }
 
   @Override
-  public void cleanup(String workspaceId) {
-    // TODO implement https://github.com/eclipse/che/issues/6767
+  public void cleanup(String workspaceId) throws InfrastructureException {
+    pvcJobExecutor.executeAsync(workspaceId, REMOVE_DIRS, getSubPath(workspaceId));
+  }
+
+  String getSubPath(String workspaceId) {
+    return workspaceId + (projectsPath.startsWith("/") ? projectsPath : '/' + projectsPath);
```

Comment:
You can normalize `projectsPath` once in the Constructor.

Reply:
+1